ui/js/compareperf.js

```diff
@@ -13,6 +13,16 @@ perf.controller('CompareChooserCtrl', [
             $scope.originalProject = $scope.newProject = projects[0];
             $scope.originalTipList = [];
             $scope.newTipList = [];
+            if ($stateParams.originalProject || $stateParams.newProject) {
```

Comment:
This `if` clause isn't what we want, since it means that we won't be able to specify orignalRevision/newRevision if orginalProject/newProject aren't specified. 

Also, lodash's findWhere is a better method of finding an individual element in an array: https://lodash.com/docs#findWhere

So instead, I'd modify the originalProject/newProject declarations above to do something like:

    $scope.originalProject = _.findWhere(projects, { name: $stateParams.originalProject }) || projects[0];